### DISCUSSION TITLE
[Question] Is websockets supported?

### DISCUSSION THREAD

**Discussion by kabaluyot** [hiero-ledger/hiero-json-rpc-relay#1201] [Q&A] [answered]
### Description

My use case is to create a a smart contract event listener API via subscriptions (https://goethereumbook.org/en/event-subscribe/). I've been seeing some WS commits and mentioned from release notes.

Do we have a public WS url in tandem with the current JSON RPC url?

### Steps to reproduce

n/a

### Hedera network

mainnet, testnet

### Version

n/a

### Operating system

macOS

**Comment by georgi-l95** [comment]
Hey @kabaluyot we are working towards support of `eth_subscribe` using websocket. We still don't have public WS url, but this will change soon.

**Comment by kabaluyot** [comment]
Hi @georgi-l95 , thanks for the response. Do you have an ETA of the feature in hashio at least on testnet? We are planning to launch a Hedera DAO paltform based from this JSON RPC Relay from Hashio .. TIA

**Comment by georgi-l95** [comment]
@kabaluyot I can't give an ETA for mainnet, but it should soon be availabe in previewnet and testnet. We have opened issue for adding the deployment details around the websocket - https://github.com/hashgraph/hedera-json-rpc-relay/issues/1166. You can follow the issue or I can ping you here, when it's available on testnet.

**Comment by kabaluyot** [comment]
Ok thanks @georgi-l95 .. Constantly checking hashio site also as well as release notes in this repo

**Comment by Nana-EC** [comment]
HI @kabaluyot as @georgi-l95 noted we're making progress towards making this available.

Please stay tuned to [JSON RPC Relay releases](https://github.com/hashgraph/hedera-json-rpc-relay/releases) where we'll announce support.

**Reply by kabaluyot** [reply]
Thank you

**Comment by Nana-EC** [comment, marked as answer]
@kabaluyot we're rolling this out.
Would appreciate you testing this on previewnet and testnet to get early feedback.
You can hit the following endpoints for `eth_chainId`, `eth_subscribe` and `eth_unsubscribe`
- previewnet - wss://previewnet.hashio.io/ws
- testnet - wss://testnet.hashio.io/ws

**Reply by Nana-EC** [reply]
Closing as we've supported this for a few sprints now.
@kabaluyot let us know if anything comes up with an issue or follow up question

**Comment by Nana-EC** [comment]
Closing this as WS support has been out for over a year now.
Thanks @kabaluyot
